Source/WTF/wtf/text/StringImpl.cpp

```diff
@@ -1417,8 +1417,8 @@ template<typename CharacterType> inline bool equalInternal(const StringImpl* a,
     if (a->length() != length)
         return false;
     if (a->is8Bit())
-        return equal(a->characters8(), b, length);
-    return equal(a->characters16(), b, length);
+        return *a->characters8() == *b && equal(a->characters8(), b, length);
```

Comment:
Maybe the equal call should be

`equal(a->characters8() + 1, b + 1, length -1)`

?

Probably not a huge difference, but might as well?

Reply:
Done.